sphinx/builders/linkcheck.py

```diff
@@ -554,6 +561,7 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     # Anchors starting with ! are ignored since they are
     # commonly used for dynamic pages
     app.add_config_value('linkcheck_anchors_ignore', ["^!"], None)
+    app.add_config_value('linkcheck_exclude_documents', [], None)
```

Comment:
Perhaps move that line next to `linkcheck_ignore`, the behavior is symmetrical to `linkcheck_ignore`, not `linkcheck_anchor_ignore` (where the link is fetched).